### ISSUE TITLE
Any plans for supporting django-oscar 2.0b1 or the 2.x?

### ISSUE THREAD

**Issue by mhoonjeon** [django-oscar/django-oscar-api#162]
I have installed django-oscar==2.0b1 and django-oscar-api==1.5.4.

A lot of things seems to be broken at the moment, and I cannot get past `python manage.py migrate` in README.

According to the django-oscar release note, the django-oscar have gone through some major backward-incompatible changes; one of which is the removal of `application` variable and using the django's default app registry instead.

Django-oscar-api uses the application in `app.py`, which consequently breaks the code. I've been fiddling around for 5 hours today to fix this, but to no avail :(

One of the things I tried was creating a new django app as in [customizing oscar api](https://django-oscar-api.readthedocs.io/en/latest/usage/customizing_oscarapi.html), and overriding the brand new `OscarConfig` class in `oscar.core.application` in 2.0b1 to replace the old `RESTApiApplication` in `oscarapi/app.py`. But it sadly didn't work.

Any plans for supporting the 2.x version? Or any idea how I could deal with this?

Thanks in advance.

**Comment by specialunderwear** [comment]
See #156

**Comment by maerteijn** [comment]
@mhoonjeon We are working on it. Expect some results and even a release the following week(s)

**Comment by mhoonjeon** [comment]
@specialunderwear @maerteijn Couldn't expect a faster response. Thanks!

**Comment by mhoonjeon** [comment]
Hi, any updates on this issue yet? 
The related PR #156 seems to be almost complete. Any plans for releases?
Thanks.

**Comment by maerteijn** [comment]
Hi @mhoonjeon ,

You could help by testing the branch and report any issues you find:
```
 pip install git+https://github.com/django-oscar/django-oscar-api@oscar-2.0
```
When no issues are found I could do a release the following week.

**Comment by mhoonjeon** [comment]
@maerteijn Okay. I will have a look and see what I can do this week. In the meantime I really appreciate what you have done! Thanks!

**Comment by mhoonjeon** [comment]
@maerteijn I having a look. But mostly seems fine. Except I found a type in the doc maybe.

The part where you explain how to add product to the basket seems to contain invalid json payload.

```python
data = {
    "url": products[1]['url'],
    "quantity": 1,
    options = [{
        "option": option_url, "value": "blue"
    }]
}

response = session.post('http://localhost:8000/api/basket/add-product/', json=data)
```

Above should probably be modified as below.

```python
data = {
    "url": products[1]['url'],
    "quantity": 1,
    "options": [{
        "option": option_url, "value": "blue"
    }]
}

response = session.post('http://localhost:8000/api/basket/add-product/', json=data)
```

[See Here](https://github.com/django-oscar/django-oscar-api/blob/oscar-2.0/docs/source/usage/communicate_with_the_api.rst#add-products-to-the-basket)

**Comment by maerteijn** [comment]
Thanks for the feedback, fixed the typo in fc670c51cd54ff1b64ac595cac317f6a240a21c8

**Comment by mhoonjeon** [comment]
I this you forgot to change "options" to key-value, which should be the right format to represent JSON. I think it should be

```
  "options": [{
        "option": option_url, "value": "blue"
    }]
```

not
```
  "options" = [{
        "option": option_url, "value": "blue"
    }]
```

**Comment by maerteijn** [comment]
Fixed your comment  and released version 2.0: https://pypi.org/project/django-oscar-api/2.0.0/
